src/Our.Umbraco.Ditto/ComponentModel/TypeConverters/DittoConverter.cs

```diff
@@ -63,9 +69,18 @@ protected virtual object ConvertMediaFromInt(int id, Type targetType, CultureInf
             // Ensure we are actually returning a media file.
             if (media.HasProperty(Constants.Conventions.Media.File))
             {
+                // If we're expecting IPublishedContent then just return
+                if (targetType == typeof(IPublishedContent))
+                    return media;
+
                 return media.As(targetType, culture);
             }
 
+
+            // If we're expecting IPublishedContent then just return
+            if (targetType == typeof(IEnumerable<IPublishedContent>))
```

Comment:
I'd like to use `IsEnumerableOfType(typeof(IPublishedContent)` here for testability and consistency.